### PR TITLE
a lot of minor changes

### PR DESCRIPTION
**Pull request by TacoTheDank**
```
updated gradles and dependencies

replaced compile with implementation, and other deprecated things

tried my hand at attempting to fix the neverending pain in the ass that is travis

updated java to 1.8 (using compileOptions, just to make sure it is (even if it was before))

converted anonymous java expressions to lambdas (and methods)

deleted redundant code usage

updated jar libraries

fixed TODO misspellings of "jontayler" to "johntaylor"

Signed-off-by: TacoTheDank <SkytkRSfan3895@gmail.com>
```
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradlew app:connectedAndroidTest` to make sure you didn't break anything

Git bash kept failed, saying 
```
"Execution failed for task ':app:connectedFdroidDebugAndroidTest'.
> com.android.builder.testing.api.DeviceException: No connected devices!"
```
Although that's not my changes that caused that, it's just that i'm an idiot and i don't know how to run a device xd. if it weren't for that, i'm sure it would've ran fine (considering i made AS rebuild the project many times over the course of these changes, with no errors in the end)

- [x] If you have multiple commits please combine them into one commit by squashing them.

if you have any doubts, here's apks I built from these changes, so you and others can test them:
https://app.box.com/s/4xk0nhdq2hycunkv6pn8qtwwkmfwkg8o

### CODE REVIEW

**Comment by TacoTheDank** [comment]
and of course travis fails lmao

File: app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelTest.java [hunk 1]
```diff
@@ -110,11 +110,9 @@ public void testSetPhoneSensorValues_phoneFlatAtLat0Long90() {
     Vector3 expectedEast = new Vector3(-1, 0, 0);
     Vector3 expectedSouth = new Vector3(0, 0, -1);
     Vector3 expectedWest = new Vector3(1, 0, 0);
-    Vector3 expectedPointing = expectedNadir;
```
Comment (jaydeetay) [inline app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelTest.java]: I'd like to keep these as they are. The variable names are meant to document their meaning.  In the case of this particular test we're showing that the "pointing" (ie the direction we're looking) is in the same direction as the nadir.

File: app/libs/README.txt [hunk 2]
```diff
@@ -1 +1 @@
-TODO(jontayler): eliminate these in favor of those from a maven repo.
\ No newline at end of file
+TODO(johntaylor): eliminate these in favor of those from a maven repo.
```
Comment (jaydeetay) [inline app/libs/README.txt]: Actually, can you leave this as-is? The misspelling is deliberate (when you're called John Taylor you have to get creative with spellings.... https://en.wikipedia.org/wiki/John_Taylor_(bass_guitarist)

File: app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelTest.java (hunk 1 above)
Comment (jaydeetay) [reply app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelTest.java]: (same elsewhere)

File: app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelWithMagneticVariationTest.java [hunk 3]
```diff
@@ -90,6 +90,13 @@ public void testAssertVectorEquals() {
     }
     assertTrue(failed);
   }
+  
+  /**
+   * Convert from degrees to radians.
+   */
+  private static float radians() {
```
Comment (jaydeetay) [inline app/src/androidTest/java/com/google/android/stardroid/test/control/AstronomerModelWithMagneticVariationTest.java]: This function needs to stay as it was with the argument.  It's converting angles in degrees to angles in radians, so it doesn't make sense to hard-code the 10 degrees.

File: app/src/fdroid/java/com/google/android/stardroid/util/Analytics.java [hunk 4]
```diff
@@ -35,8 +35,8 @@ public void setEnabled(boolean enabled) { }
    * Tracks a screen view.
    */
   public void trackPageView(String page) { }
-
-  /**
+  
+  /*
```
Comment (jaydeetay) [inline app/src/fdroid/java/com/google/android/stardroid/util/Analytics.java]: I think this was correct - it's supposed to be javadoc.

File: app/src/main/java/com/google/android/stardroid/activities/DiagnosticActivity.java [hunk 5]
```diff
@@ -240,9 +240,7 @@ public void onSensorChanged(SensorEvent event) {
             valuesViewId = R.id.diagnose_rotation_matrix_row3_txt;
         }
         float[] rowValues = new float[3];
-        for (int col = 0; col < 3; ++col) {
-          rowValues[col] = matrix[row * 3 + col];
-        }
+        System.arraycopy(matrix, row * 3 + 0, rowValues, 0, 3);
```
Comment (jaydeetay) [inline app/src/main/java/com/google/android/stardroid/activities/DiagnosticActivity.java]: Could drop the +0 ?

File: app/libs/README.txt (hunk 2 above)
Comment (TacoTheDank) [reply app/libs/README.txt]: lol how are people actually mistaking you for the founder of Duran Duran?

**Comment by TacoTheDank** [comment]
ill redo this merge request

**Pull request closed without merging**